src/signals.jl

```diff
@@ -68,6 +68,17 @@ const SamplesInfo = @row("onda.samples-info@1",
                          sample_type::AbstractString = onda_sample_type_from_julia_type(sample_type),
                          sample_rate::LPCM_SAMPLE_TYPE_UNION = convert_number_to_lpcm_sample_type(sample_rate))
 
+
+"""
+    Base.copy(s::SamplesInfo)
+
+Create a new `SamplesInfo` by performing a shallow `copy` of each of its fields, if possible according to `Base.applicable`.
```

Comment:
```suggestion
Call `copy` on each field of the given `SamplesInfo` for which an applicable `copy` method
exists and return the resulting `SamplesInfo`.
```
IMO the fact that `applicable` is used should be kept an implementation detail. It would be weird to me if we had to change the docstring if we decided to switch the determination to use `hasmethod`.